communicator/ssh/communicator.go

```diff
@@ -59,6 +61,39 @@ func New(address string, config *Config) (result *comm, err error) {
 	return
 }
 
+// ConnectToSSH returns the same data as New() but will retry the connection if
+// there is a problem. Will retry after [timeout] seconds up to [retry] times.
+func ConnectToSSH(address string, config *Config, timeout uint8, retry uint8) (communicator *comm, err error) {
```

Comment:
It'd be more idiomatic to just make the `timeout` field a `time.Duration`.

I'd also remove the `retry` field. The SSH connection step in `helper/communicator` automatically will retry the entire `New` process so you don't need to retry within it. That'll simplify this  a ton too.